pallets/messages/src/tests.rs

```diff
@@ -270,12 +270,13 @@ fn get_messages_by_schema_with_valid_request_should_return_paginated() {
 		assert_eq!(
 			pagination_response.content[0],
 			MessageResponse {
-				msa_id: 10,
-				payload: payload.clone(),
+				msa_id: Some(10),
+				payload: Some(payload.clone()),
 				index: from_index as u16,
 				provider_msa_id: 1,
 				block_number: 0,
-				payload_length: payload.len().try_into().unwrap(),
+				payload_length: None,
```

Comment:
Also in my opinion this payload length should not be None, it should be the length of the actual payload/message.

Reply:
We talked about that back and forth in planning. In the end the choice was that payload_length was just for the off-chain payloads as otherwise it was just extra data

Reply:
I recall, I'm just second guessing.  Is it so much extra data that it's worth the inconsistency and extra logic? I'd prefer the consistency but if it's really outweighed by storage cost then it sounds like we just have to deal with it.